src/vivarium_csu_ltbi/components/treatment.py

```diff
@@ -0,0 +1,139 @@
+                                                                    key_columns=['sex'],
+                                                                    value_columns=['value'])
+        return coverage_table_with_hiv, coverage_table_under_five_hhtb
+
+    def get_coverage(self, index):
+        pop = self.population_view.get(index)
+
+        # TODO: Switch to a dataframe of zeroes with index = index, overwrite
+        #  with the below
+        coverage = pd.DataFrame(data={'3HP': 0.0, '6H': 0.0, 'untreated': 1.0},
+                                index=pop.index)
+
+        # Generate bit masks for our group conditions
+        with_hiv = self.get_hiv_positive_subgroup(pop)
+        under_five_hhtb = self.get_under_five_hhtb_subgroup(pop)
+
+        # HIV+ Group
+        with_hiv_idx = pop.loc[with_hiv].index
+        with_hiv_threehp = self.three_hp_with_hiv(with_hiv_idx)
+        with_hiv_sixh = self.six_h_with_hiv(with_hiv_idx)
+
+        coverage.loc[with_hiv_idx, '3HP'] = with_hiv_threehp
+        coverage.loc[with_hiv_idx, '6H'] = with_hiv_sixh
+        coverage.loc[with_hiv_idx, 'untreated'] = 1. - (with_hiv_threehp + with_hiv_sixh)
+
+        # Under 5 and exposed to household TB group
+        under_five_hhtb_idx = pop.loc[under_five_hhtb].index
+        under_five_hhtb_threehp = self.three_hp_under_five_hhtb(under_five_hhtb_idx)
+        under_five_hhtb_sixh = self.six_h_under_five_hhtb(under_five_hhtb_idx)
+
+        coverage.loc[under_five_hhtb_idx, '3HP'] = under_five_hhtb_threehp
+        coverage.loc[under_five_hhtb_idx, '6H'] = under_five_hhtb_sixh
+        coverage.loc[under_five_hhtb_idx, 'untreated'] = 1. - (under_five_hhtb_threehp + under_five_hhtb_sixh)
+
+        # In both groups
+        in_both_groups = with_hiv & under_five_hhtb
+        coverage.loc[in_both_groups, '3HP'] = 1. - ((1. - with_hiv_threehp.loc[in_both_groups]) * (1. - under_five_hhtb_threehp.loc[in_both_groups]))
+        coverage.loc[in_both_groups, '6H'] = 1. - ((1. - with_hiv_sixh.loc[in_both_groups]) * (1. - under_five_hhtb_sixh.loc[in_both_groups]))
+        coverage.loc[in_both_groups, 'untreated'] = 1 - (coverage.loc[in_both_groups, '3HP'] + coverage.loc[in_both_groups, '6H'])
+
+        return coverage
+
+    def enforce_not_eligible(self, data, timestep):
+        pop = self.population_view.get(data.index)
+
+        with_hiv = self.get_hiv_positive_subgroup(pop)
+        under_five_hhtb = self.get_under_five_hhtb_subgroup(pop)
+
+        data.loc[~with_hiv & ~under_five_hhtb, '3HP'] = 0.0
+        data.loc[~with_hiv & ~under_five_hhtb, '6H'] = 0.0
+        data.loc[~with_hiv & ~under_five_hhtb, 'untreated'] = 1.0
+
+        return data
+
+    def on_initialize_simulants(self, pop_data):
+        initialized = pd.DataFrame({'treatment_date': pd.NaT,
+                                    'treatment_type': 'untreated'}, index=pop_data.index)
+        self.population_view.update(initialized)
+
+    def on_time_step(self, event):
```

Comment:
Use time step prepare, use current clock.  This bootstraps the population initialization.